src/service/publish-service.js

```diff
@@ -4,76 +4,62 @@ const {
     NETWORK_PROTOCOLS,
     HANDLER_ID_STATUS,
     PUBLISH_REQUEST_STATUS,
+    PUBLISH_STATUS,
 } = require('../constants/constants');
 
 const mutex = new Mutex();
 
 class PublishService {
     constructor(ctx) {
         this.logger = ctx.logger;
-
+        this.config = ctx.config;
         this.repositoryModuleManager = ctx.repositoryModuleManager;
         this.commandExecutor = ctx.commandExecutor;
         this.networkModuleManager = ctx.networkModuleManager;
         this.handlerIdService = ctx.handlerIdService;
     }
 
-    async processPublishResponse(command, status, errorMessage = null) {
-        const { handlerId, ual, assertionId, numberOfFoundNodes } = command.data;
+    async processPublishResponse(command, responseStatus, errorMessage = null) {
+        const { handlerId, ual, assertionId, numberOfFoundNodes, leftoverNodes } = command.data;
 
         const self = this;
-        let numberOfResponses = 0;
+        let responses = 0;
+        let failedNumber = 0;
+        let completedNumber = 0;
         await mutex.runExclusive(async () => {
             await self.repositoryModuleManager.createPublishResponseRecord(
-                status,
+                responseStatus,
                 handlerId,
                 errorMessage,
             );
 
-            numberOfResponses = await self.repositoryModuleManager.getNumberOfPublishResponses(
-                handlerId,
-            );
+            responses = await self.repositoryModuleManager.getPublishResponsesStatuses(handlerId);
         });
 
-        if (numberOfFoundNodes === numberOfResponses) {
-            this.logger.info(`Finalizing publish for handlerId: ${handlerId}`);
-
-            await this.handlerIdService.cacheHandlerIdData(handlerId, { ual, assertionId });
-
-            await this.handlerIdService.updateHandlerIdStatus(
-                handlerId,
-                HANDLER_ID_STATUS.PUBLISH.PUBLISH_REPLICATE_END,
-            );
-            await this.handlerIdService.updateHandlerIdStatus(
-                handlerId,
-                HANDLER_ID_STATUS.PUBLISH.PUBLISH_END,
-            );
-
-            const responseStatuses = await this.repositoryModuleManager.getPublishResponsesStatuses(
-                handlerId,
-            );
-            let failedNumber = 0;
-            let completedNumber = 0;
-
-            responseStatuses.forEach((responseStatus) => {
-                if (responseStatus === PUBLISH_REQUEST_STATUS.FAILED) {
-                    failedNumber += 1;
-                } else {
-                    completedNumber += 1;
-                }
-            });
-
+        responses.forEach((response) => {
+            if (response === PUBLISH_REQUEST_STATUS.FAILED) {
```

Comment:
I think it should be response.status here

Reply:
No it's returning as string array

Reply:
You were right, i've added the change